File: Plugins/Aspose_Tasks_Java_for_Ruby/lib/asposetasksjava/ResourceAssignments/stopandresumeassignment.rb
module Asposetasksjava
  module StopAndResumeAssignment
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'
                
        # Instantiate project object
        project = Rjb::import('com.aspose.tasks.Project').new(data_dir + 'test_tasks.mpp')

        resource_assignments = project.getResourceAssignments().toList()

        # Parse through all the collected resource assignments
        i = 0
        while i < resource_assignments.size()
          ra = resource_assignments.get(i)
          if ra.get(Rjb::import('com.aspose.tasks.Asn').STOP).toString() == "1/1/2000"
            puts "NA"
          else
             puts ra.get(Rjb::import('com.aspose.tasks.Asn').STOP).toString()
          end

          if ra.get(Rjb::import('com.aspose.tasks.Asn').RESUME).toString() == "1/1/2000"
            puts "NA"
          else
            puts ra.get(Rjb::import('com.aspose.tasks.Asn').RESUME).toString()
          end
          i +=1
        end    
    end
  end
end

File: Plugins/Aspose_Tasks_Java_for_PHP/src/aspose/tasks/WorkingWithResourceAssignments/StopAndResumeAssignment.php
<?php
namespace Aspose\Tasks\WorkingWithResourceAssignments;

use com\aspose\tasks\Project as Project;
use com\aspose\tasks\Asn as Asn;
class StopAndResumeAssignment{

    public static function run($dataDir=null){

        # Instantiate project object
        $project = new Project($dataDir . 'test_tasks.mpp');

        $resource_assignments = $project->getResourceAssignments()->toList();

        # Parse through all the collected resource assignments
        $i = 0;
        $asn=new Asn();
        while ($i < sizeof($resource_assignments)) {
          $ra = $resource_assignments -> get($i);
          if ((string)$ra -> get($asn -> STOP) == "1/1/2000") {
              print "NA".PHP_EOL;
              }
          else {
              print (string)$ra -> get($asn -> STOP);// . toString();
          }

          if ((string)$ra -> get($asn -> RESUME) == "1/1/2000"){
          print "NA".PHP_EOL;
          }
          else {
              print (string)$ra -> get($asn -> RESUME);// . toString()
          }
          $i += 1;
        }


    }
}
?>